File: SSS/biblioteca/src/main/java/com/biblio/biblioteca/Entity/libro.java
package com.biblio.biblioteca.Entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "libro")
public class libro {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@Column(name = "titulo", nullable = false, unique = true, length = 50)
	private String titulo;
	
	@Column(name = "autor", nullable = false, unique = false, length = 50)
	private String autor;
	
	@Column(name = "isbn", nullable = false, unique = true, length = 50)
	private String isbn;

	@Column(name = "genero", nullable = false, unique = false, length = 50)
	private String genero;
	
	@Column(name = "ejemplares_disponibles", nullable = false, unique = false, length = 50)
	private int ejemplaresDisponibles;
	
	@Column(name = "ejemplares_ocupados", nullable = false, unique = false, length = 50)
	private int ejemplaresOcupados;
	
	@Column(name = "estado", nullable = false, unique = false)
	private boolean estado;


	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getTitulo() {
		return titulo;
	}

	public void setTitulo(String titulo) {
		this.titulo = titulo;
	}

	public String getAutor() {
		return autor;
	}

	public void setAutor(String autor) {
		this.autor = autor;
	}

	public String getIsbn() {
		return isbn;
	}

	public void setIsbn(String isbn) {
		this.isbn = isbn;
	}

	public String getGenero() {
		return genero;
	}

	public void setGenero(String genero) {
		this.genero = genero;
	}

	public int getEjemplaresDisponibles() {
		return ejemplaresDisponibles;
	}

	public void setEjemplaresDisponibles(int ejemplaresDisponibles) {
		this.ejemplaresDisponibles = ejemplaresDisponibles;
	}

	public int getEjemplaresOcupados() {
		return ejemplaresOcupados;
	}

	public void setEjemplaresOcupados(int ejemplaresOcupados) {
		this.ejemplaresOcupados = ejemplaresOcupados;
	}

	public boolean isEstado() {
		return estado;
	}

	public void setEstado(boolean estado) {
		this.estado = estado;
	}




}

File: biblioteca_prueba/lib/models/libro.dart
// lib/models/libro.dart

class Libro {
  int id;
  String titulo;
  String autor;
  String isbn;
  String genero;
  int ejemplaresDisponibles;
  int ejemplaresOcupados;
  String estado;

  Libro({
    required this.id,
    required this.titulo,
    required this.autor,
    required this.isbn,
    required this.genero,
    required this.ejemplaresDisponibles,
    required this.ejemplaresOcupados,
    required this.estado,
  });

  factory Libro.fromMap(Map<String, dynamic> json) {
    return Libro(
      id: json['id'],
      titulo: json['titulo'],
      autor: json['autor'],
      isbn: json['isbn'],
      genero: json['genero'],
      ejemplaresDisponibles: json['ejemplaresDisponibles'],
      ejemplaresOcupados: json['ejemplaresOcupados'],
      estado: json['estado'],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'titulo': titulo,
      'autor': autor,
      'isbn': isbn,
      'genero': genero,
      'ejemplaresDisponibles': ejemplaresDisponibles,
      'ejemplaresOcupados': ejemplaresOcupados,
      'estado': estado,
    };
  }
}
